core/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/AbstractHistogramBuilder.java

```diff
@@ -21,86 +21,50 @@
 
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
-import org.elasticsearch.common.rounding.Rounding;
 import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.search.aggregations.InternalAggregation.Type;
+import org.elasticsearch.search.aggregations.support.ValueType;
 import org.elasticsearch.search.aggregations.support.ValuesSource;
 import org.elasticsearch.search.aggregations.support.ValuesSourceAggregationBuilder;
 import org.elasticsearch.search.aggregations.support.ValuesSourceType;
 
 import java.io.IOException;
 import java.util.Objects;
 
-public abstract class AbstractHistogramBuilder<AB extends AbstractHistogramBuilder<AB>>
+// only exists to share some code between the two histogram builders
+// don't make it public
+abstract class AbstractHistogramBuilder<AB extends AbstractHistogramBuilder<AB>>
         extends ValuesSourceAggregationBuilder<ValuesSource.Numeric, AB> {
```

Comment:
ok, but since its abstract can we still document what its doing? Does it really require the insane generic typing?